### PR TITLE
[SCC-3597] Initialize production and development docker config

### PR DESCRIPTION
**Pull request by dgcohen**
This PR initializes the Docker configuration for the new research catalog app.

To run locally:
cd into research-catalog
run `docker-compose up --build --force-recreate`

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: '1'
```
Comment (EdwinGuzman) [inline docker-compose.yml:1]: I get this error locally:
Screen Shot 2023-06-30 at 9 51 28 AM

Try version `'2'`, that works for me. Looks like v1 won't be supported anymore https://docs.docker.com/compose/compose-file/compose-versioning/

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (EdwinGuzman) [reply docker-compose.yml:1]: Or version 3. I only just tried 2 but maybe 3 is the way to go.

Comment (dgcohen) [reply docker-compose.yml]: Good call! Updated to version 3.

**Review by EdwinGuzman — APPROVED** [review]

**Review by nonword — APPROVED** [review]
Looks good to me!

**Pull request merged**